module/Activity/src/Activity/Service/Activity.php

```diff
@@ -197,9 +222,54 @@ public function reset(ActivityModel $activity)
      */
     public function disapprove(ActivityModel $activity)
     {
+        $this->permissionManageActivityOrException();
+
         $activity->setStatus(ActivityModel::STATUS_DISAPPROVED);
         $em = $this->getServiceManager()->get('Doctrine\ORM\EntityManager');
         $em->persist($activity);
         $em->flush();
     }
+
+    /*
+     * Some general ACL functions
+     */
+
+    /**
+     * Checks if the user is allowed to view the activity, and if not, throws an used permission exception
+     */
+    public function permissionViewActivityOrException()
```

Comment:
I could do that, however I think that permissions should not be too granular (It hurts the simplicity of the acl), the approve/disapprove/reset actions are actions are quite closely related I think.

Also what do you think of mu using other methods to do the permission check. In other Services, e.g. (https://github.com/GEWIS/gewisweb/blob/master/module/Photo/src/Photo/Service/Photo.php) this is not done, which imo both clutters the code inside the function, and is anything but DRY.